File: TP2/Exo 2.5/python/bd_ecole.py
# Classe Etudiant
class Etudiant:
    def __init__(self, nom, prenom, adresse, note1, note2):
        # Initialisation des attributs de la classe
        self.Nom = nom
        self.Prenom = prenom
        self.Adresse = adresse
        # Les notes sont stockées sous forme de liste de float
        self.Notes = [float(note1), float(note2)]

    # Méthode pour afficher les détails de l'étudiant
    def afficherEtudiant(self):
        print(f"Nom : {self.Nom}")
        print(f"Prénom : {self.Prenom}")
        print(f"Adresse : {self.Adresse}")
        print(f"Note 1 : {self.Notes[0]}")
        print(f"Note 2 : {self.Notes[1]}")

# Fonction pour ajouter un nouvel étudiant
def ajouterEtudiant():
    print(f"Saisissez les informations de l'étudiant :")
    nom = input("Nom : ")
    prenom = input("Prénom : ")
    adresse = input("Adresse : ")
    note1 = float(input("Note 1 : "))  # Lecture de la première note
    note2 = float(input("Note 2 : "))  # Lecture de la deuxième note

    # Retourne une nouvelle instance de la classe Etudiant
    return Etudiant(nom, prenom, adresse, note1, note2)

def main():
    etudiants = []  # Liste pour stocker les instances de la classe Etudiant

    # Ajout de 5 étudiants
    for i in range(5):
        print(f"\nÉtudiant {i + 1}:")
        etudiant = ajouterEtudiant()  # Création d'un nouvel étudiant
        etudiants.append(etudiant)  # Ajout de l'étudiant à la liste

    # Affichage des informations des étudiants
    print("\nInformations des étudiants :\n")
    for etudiant in etudiants:
        etudiant.afficherEtudiant()  # Appel de la méthode pour afficher les détails de chaque étudiant
        print()  # Ligne vide pour séparer les informations des étudiants

# Point d'entrée du programme
if __name__ == "__main__":
    main()

File: TP2/Exo 2.5/C/bd_ecole.c
#include <stdio.h>
#include <string.h>

// Définition de la structure pour stocker les informations d'un étudiant
struct Etudiant 
{
    char Nom[50];
    char Prenom[50];
    char Adresse[100];
    float Note[2];
};

// Fonction pour ajouter les informations d'un étudiant
void ajouterEtudiant(struct Etudiant *etudiant) {
    char buffer[100];

    // Lecture et copie du nom
    printf("Nom : ");
    scanf("%s", buffer);
    strcpy(etudiant->Nom, buffer); // Utilisation de strcpy pour copier le nom

    // Lecture et copie du prénom
    printf("Prénom : ");
    scanf("%s", buffer);
    strcpy(etudiant->Prenom, buffer); // Utilisation de strcpy pour copier le prénom

    // Lecture et copie de l'adresse
    printf("Adresse : ");
    getchar(); // Consomme le caractère '\n' laissé par scanf
    fgets(buffer, 100, stdin); // Lecture de l'adresse (qui peut contenir des espaces)
    buffer[strcspn(buffer, "\n")] = 0; // Supprime le caractère '\n' en fin de chaîne
    strcpy(etudiant->Adresse, buffer); // Utilisation de strcpy pour copier l'adresse

    // Lecture des notes
    printf("Note 1 : ");
    scanf("%f", &etudiant->Note[0]);

    printf("Note 2 : ");
    scanf("%f", &etudiant->Note[1]);
}

// Fonction pour afficher les informations d'un étudiant
void afficherEtudiant(struct Etudiant etudiant)
{
    printf("Nom : %s\n", etudiant.Nom);
    printf("Prénom : %s\n", etudiant.Prenom);
    printf("Adresse : %s\n", etudiant.Adresse);
    printf("Note 1 : %.2f\n", etudiant.Note[0]);
    printf("Note 2 : %.2f\n", etudiant.Note[1]);
}

int main()
{
    struct Etudiant etudiants[5]; // Tableau de 5 étudiants

    // Ajout des informations pour 5 étudiants
    for (int i = 0; i < 5; i++)
    {
        printf("Étudiant %d :\n", i + 1);
        ajouterEtudiant(&etudiants[i]);
        printf("\n");
    }
    
    // Affichage des informations pour chaque étudiant
    printf("Informations des étudiants :\n\n");
    for (int j = 0; j < 5; j++)
    {
        printf("Étudiant %d :\n", j + 1);
        afficherEtudiant(etudiants[j]);
        printf("\n");
    }
    
    return 0;
}
